modules/mono/glue/cs_files/Point2.cs

```diff
@@ -0,0 +1,419 @@
+using System;
+using System.Runtime.InteropServices;
+
+#if REAL_T_IS_DOUBLE
+using real_t = System.Double;
+#else
+using real_t = System.Single;
+#endif
+
+namespace Godot
+{
+    /// <summary>
+    /// Point2 is a class similar to Vector2, except with integers.
+    /// It is useful, for example, when working with points on a 2D grid.
+    /// Not all concepts of Vector2 apply, such as normalization and any functions that depend on normalization. 
+    /// However, new concepts can exist with Point2 such as bitwise operations.
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential)]
+    public struct Point2 : IEquatable<Point2>
+    {
+        public enum Axis
+        {
+            X = 0,
+            Y
+        }
+
+        public int x;
+        public int y;
+
+        public int this[int index]
+        {
+            get
+            {
+                switch (index)
+                {
+                    case 0:
+                        return x;
+                    case 1:
+                        return y;
+                    default:
+                        throw new IndexOutOfRangeException();
+                }
+            }
+            set
+            {
+                switch (index)
+                {
+                    case 0:
+                        x = value;
+                        return;
+                    case 1:
+                        y = value;
+                        return;
+                    default:
+                        throw new IndexOutOfRangeException();
+                }
+            }
+        }
+
+        public Point2 Abs()
+        {
+            return new Point2( Mathf.RoundToInt(Mathf.Abs(x)), Mathf.RoundToInt(Mathf.Abs(y)) );
```

Comment:
Could be changed to <code>return new Point2( Mathf.Abs(x), Mathf.Abs(y) )</code> if you create Math.Abs for integers.